README.md

```diff
@@ -1,29 +1,310 @@
+
+### Targeting
+
+Sometimes, the value of a flag must consider some dynamic criteria about the application or user, such as the user's location, IP, email address, or the server's location.
+In OpenFeature, we refer to this as [targeting](https://openfeature.dev/specification/glossary#targeting).
+If the flag management system you're using supports targeting, you can provide the input data using the [evaluation context](https://openfeature.dev/docs/reference/concepts/evaluation-context).
+
+```rust
+// Create a global evaluation context and set it into the API.
+// Note that this is optional. By default it uses an empty one.
+let mut api = OpenFeature::singleton_mut().await;
+api.set_evaluation_context(global_evaluation_context).await;
+
+// Set client level evaluation context.
+// It will overwrite the global one for the existing keys.
+let mut client = api.create_client();
+client.set_evaluation_context(client_evaluation_context);
+
+// Pass evaluation context in evaluation functions.
+// This one will overwrite the globla evaluation context and 
+// the client level one.
+client.get_int_value("flag", &evaluation_context, None);
+```
+
+### Hooks
+
+[Hooks](https://openfeature.dev/docs/reference/concepts/hooks) allow for custom logic to be added at well-defined points of the flag evaluation life-cycle.
+Look [here](https://openfeature.dev/ecosystem/?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Hook&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=Rust) for a complete list of available hooks.
+If the hook you're looking for hasn't been created yet, see the [develop a hook](#develop-a-hook) section to learn how to build it yourself.
+
+Once you've added a hook as a dependency, it can be registered at the global, client, or flag invocation level.
+
+<!-- TODO: code example of setting hooks at all levels -->
+
+### Logging
+
+<!-- TODO: talk about logging config, and code example -->
+
+### Named clients
+
+Clients can be given a name.
+A name is a logical identifier which can be used to associate clients with a particular provider.
+If a name has no associated provider, the global provider is used.
+
+```rust
+// Create a named provider and bind it.
+api.set_named_provider(
+    "named",
+    NoOpProvider::builder().int_value(42).build())
+.await;
+
+// This named client will use the feature provider bound to this name.
+let client = api.create_named_client("named");
+
+assert_eq!(client.get_int_value("key", None, None).await.unwrap(), 42);
+```
+
+### Eventing
+
+Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
+Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
+Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
+
+Please refer to the documentation of the provider you're using to see what events are supported.
```

Comment:
This section should be commented out until events are supported.

Reply:
Done. Thank you for pointing it out!